python/lsst/obs/base/instrument.py

```diff
@@ -169,6 +169,22 @@ def fromName(name: str, registry: Registry) -> Instrument:
         instrument = doImport(cls)
         return instrument()
 
+    @staticmethod
+    def importAll(registry: Registry) -> None:
+        """Import all the instruments known to this registry.
+
+        This will ensure that all metadata translators have been registered.
+
+        Parameters
+        ----------
+        registry : `lsst.daf.butler.Registry`
+            Butler registry to query to find the information.
+        """
+        dimensions = list(registry.queryDimensions("instrument"))
+        for dim in dimensions:
+            cls = dim.records["instrument"].class_name
+            doImport(cls)
```

Comment:
I think we should wrap this in a try block, _maybe_ with a warning.  We don't want to require all obs packages to be setup to use any repo in which their instruments were once registered.

Reply:
Okay. You have a point that in theory someone could set up obs_lsst and not obs_decam.